### ISSUE TITLE
False positive on @Environment(\.presentationMode)

### ISSUE THREAD

**Issue by chrisngabp** [realm/SwiftLint#5103]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The line `@Environment(\.presentationMode) private var presentationMode` is complaining with an attributes violation: `Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)
Done linting! Found 1 violation, 0 serious in 1639 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.52.3`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `CocoaPods`
* Which Xcode version are you using (check `xcodebuild -version`)? `14.3`

```swift
// This triggers a violation:
@Environment(\.presentationMode) private var presentationMode
```

**Comment by SimplyDanny** [comment]
The rule has an option `attributes_with_arguments_always_on_line_above` which is `true` by default.

**Comment by chrisngabp** [comment]
@SimplyDanny maybe we can improve the text `Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)` to be `Attributes should be on their own lines in functions, variables with attributes and types, but on the same line as variables and imports (attributes)`
